### PR TITLE
Add separate blog page

### PR DESCRIPTION
**Pull request by Wodann**
This PR focuses on cleaning up the frontpage by reducing the number of displayed blogposts, instead opting for a separate blog page that lists all entries.

This is a first step which will soon be followed by adding some more WG-related information to improve on-boarding for the WG's meetings and Github Issues. I thought it better to split PRs so people can review them individually.

### CODE REVIEW

**Comment by 17cupsofcoffee** [comment]
I'm for the general idea of this PR (the front page is going to get very long if we don't split stuff up a bit!), but I do have a few things that I think need resolving first:

* Changing the URLs will break **all** of the existing links to the site on Twitter and Reddit, which wouldn't be great. Either the URLs should stay the same, or you'll need to go through and add aliases for each existing post so that Zola will generate redirects.
    * Similarly, please don't remove all of the existing aliases!
* I don't think adding `transparent = true` to a post actually does anything - Zola's docs only mention it in the context of a section (`_index.md`) file. I might be wrong though.
* Could the markup for post lists be extracted out into a [macro](https://tera.netlify.com/docs/#macros) so it's not duplicated in multiple pages?
    * If you're not familiar enough with Tera/Zola to do this, it's fine - I can pick it up as a seperate PR.

Also, a couple of styling nitpicks:

* I think the `@import "./minima"` that's been added is redundant - the root file already brings it into scope. The `inline-block` on `.post-link` also doesn't seem to do anything.
* [`safe` is special-cased by Tera, so you have to use it as the last filter in the chain](https://tera.netlify.com/docs/#safe) - it needs to look like this, otherwise you'll lose some of the apostrophes and stuff in the output:
    * `{{ post.description | default(value=post.content | striptags | trim | truncate(length=max_length)) | safe }}`

**Comment by 17cupsofcoffee** [comment]
Also, here's some screenshots if other people don't want to build the branch locally:

image


image

**Comment by Wodann** [comment]
@17cupsofcoffee Thanks for the review. To answer your questions:

> Changing the URLs will break all of the existing links to the site on Twitter and Reddit, which wouldn't be great. Either the URLs should stay the same, or you'll need to go through and add aliases for each existing post so that Zola will generate redirects.
> Similarly, please don't remove all of the existing aliases!

Sorry, that's my bad. I tried using the alias url's locally, but they weren't directing me anywhere, so thought it might be incorrect. I will fix them and make sure that the links still work by trying some of RustGamedev's tweets.

> I don't think adding transparent = true to a post actually does anything - Zola's docs only mention it in the context of a section (_index.md) file. I might be wrong though.

It actually does. Since the newsletters are all grouped in folders, when you create a new `section` and request `{% for post in section.pages %}`, it won't actually include the `_index.md` files in the subdirectories unless `transparent = true`.

> Could the markup for post lists be extracted out into a [macro](https://tera.netlify.com/docs/#macros) so it's not duplicated in multiple pages?

I've never used tera until now, but I'm figuring it out so far. Will give it a try.

> I think the @import "./minima" that's been added is redundant - the root file already brings it into scope.

I'll see if I can remove it.

> The inline-block on .post-link also doesn't seem to do anything.

There is actually a difference. `inline-block` will limit the clicking rectangle to the width and height of the text lines, whereas `block` will occupy the full width of the parent container. So on mobile you still get the nice clicking experience, while on tablet/PC you won't be directed to another page just because you clicked on a piece of whitespace far to the right from the title/link.

inline-block:
image

block:
image

> safe is special-cased by Tera, so you have to use it as the last filter in the chain - it needs to look like this, otherwise you'll lose some of the apostrophes and stuff in the output:
`{{ post.description | default(value=post.content | striptags | trim | truncate(length=max_length)) | safe }}`

The documentation I read mentioned [preceding `striptags` with `safe`](https://tera.netlify.com/docs/#striptags).

**Comment by Wodann** [comment]
@17cupsofcoffee I'm pretty sure I addressed all of your requested changes in the above three commits. If/when you are happy with them, I can `git rebase` the commits into the original changes to make history concise.

**Comment by 17cupsofcoffee** [comment]
Cool, I'll take a look at the changes now :)

> It actually does. Since the newsletters are all grouped in folders, when you create a new section and request {% for post in section.pages %}, it won't actually include the _index.md files in the subdirectories unless transparent = true.

Ah, I thought that only affected `_index.md` files, not `index.md`. I'll take your word for it :)

> There is actually a difference.

D'oh, you're right. Ignore me :)

File: content/blog/2019-08-18-introducing-the-rust-game-development-working-group.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 +++
 title = "Introducing the Rust Game Development Working Group"
-aliases = ["/2019/08/18/introducing-the-rust-game-development-working-group", "/2019/08/18/introducing-the-rust-game-development-working-group.html"]
```
Comment (17cupsofcoffee) [inline content/blog/2019-08-18-introducing-the-rust-game-development-working-group.md:3]: Please can you add these 2 aliases back to the list? They're there for compatibility with the old Jekyll URLs.

Comment (Wodann) [reply content/blog/2019-08-18-introducing-the-rust-game-development-working-group.md:3]: Oops, missed that one.

**Comment by Wodann** [comment]
> (I still don't know if I think the change from /posts to /blog is entirely neccecary - I think I'm missing the context of what extra content you/the WG are hoping to add in future, though.)

The reason I did that is because I added the `/blog` page that lists all blogs. That naming made most sense to me. At that point you cannot just `redirect` to the `/posts` to reach all md articles underneath that namespace - I think.

I can also just put everything under `/posts`, but it seemed to make most sense if people can just type `rust-gamedev.github.io/blog`.

The bigger change that I was planning was to add some information on how people can contribute to the WG and where to find appropriate information on the home page. Now you have to go through 3 or more layers before you actually find the necessary information, at which point - I am afraid - you will already have lost a lot of potential contributors 😅

**Comment by 17cupsofcoffee** [comment]
Ah, that makes sense, I definitely like the idea of there being some contributing info/extra links on the homepage.

> I can also just put everything under /posts, but it seemed to make most sense if people can just type rust-gamedev.github.io/blog.

I would be fine with either option - let's see what the other people tagged in think 🙂

**Comment by Wodann** [comment]
I just resolved your only remaining comment. Let's wait and see what the others think. Then I'll fix problems accordingly and rebase to clean up the history a bit.

**Comment by Wodann** [comment]
@ozkriff just checking to see whether you had any additional comments?

File: content/blog/newsletter-006/index.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 +++
 title = "This Month in Rust GameDev #6 - January 2020"
 date = 2020-02-16
+description = "The start of the new year hails the arrival of a slur of new content updates, job openings, and released talks. Dig into this sixth issue of the Rust GameDev Workgroup's monthly newsletter for your dose of Rust gamedev goodieness."
```
Comment (ozkriff) [inline content/blog/newsletter-006/index.md:4]: Is it possible to use multiline strings here to comply with the 80 chars limit?

Comment (Wodann) [reply content/blog/newsletter-006/index.md:4]: I'll check to see if that still displays correctly

**Comment by Wodann** [comment]
> I don't really feel that section renaming and summaries are worth it, but I'm fine with these changes.

Would you rather only show titles?

**Comment by ozkriff** [comment]
> Would you rather only show titles?

@Wodann Initially I decided not to use summaries for the newsletter because it seems to me that every issue has too many changes from too many projects to meaningfully summarise all that in any other way except for rephrasing "Here're some updates about rust games, libraries, and other tools".

**Comment by Wodann** [comment]
@ozkriff @17cupsofcoffee After having a discussion with @ozkriff I decide to greatly minimize the set of changes. I also reverted back to usage of `posts` instead of `blog` just to simplify this PR. 

The PR now merely minimises styling on the front page by only listing the latest **three** posts and adds a button that redirects to a complete listing of all posts. For styling we might want to consider doing something [similar to Rust](https://blog.rust-lang.org) - in the future.

**Comment by Wodann** [comment]
For reference, this is how the homepage looks now:

image


And this is the complete listing of all posts - on a separate page:

image

**Comment by ozkriff** [comment]
:timer_clock: Merging this tomorrow if there're no objections

**Comment by Wodann** [comment]
> One last tweak that might be nice to do before merging - could you add transparent = true to the newsletter template in the root of the directory?

Well-spotted. I'll fix that.

> I wouldn't be opposed to styling changes if someone came up with a good design.

Nor would I. Design just never came as naturally to me as programming 😬

**Comment by Wodann** [comment]
I added `transparent = true` to the newsletter template. The PR should be ready to merge now.

**Comment by ozkriff** [comment]
Merged!

**Pull request merged**